### ISSUE TITLE
[Feature Request] Bring masks back

### ISSUE THREAD

**Issue by andreixk** [vuetifyjs/vuetify#10890]
### Problem to solve
Vuetify 1.5 used to have this functionality implemented properly. Current proposition of using 3rd party plugins (especially vue-the-mask) is useless. Those plugins are clumsy, incomplete and generally require a lot of re-work to integrate. each v-text-field becomes a 10-15 line monstrosity, forms are a nightmare to create and even worse to debug.

### Proposed solution
Simple: bring back the masks that were in v1.5

**Comment by Mikilll94** [comment]
Yes, It would be great to bring mask functionality again to Vuetify 2.x

About [vue-the-mask](https://github.com/vuejs-tips/vue-the-mask) I will add also that:
- This plugin is outdated. Latest version was released on 10 Oct 2017. There are many bugs in this plugin which were not fixed.
Just look here:
https://github.com/vuejs-tips/vue-the-mask/issues/137
https://github.com/vuejs-tips/vue-the-mask/issues/112

- This plugin has no MIT license, so theoretically you cannot use it.
https://github.com/vuejs-tips/vue-the-mask/issues/111

**Comment by MajesticPotatoe** [comment]
We have plans to bring this back for v3. The likelihood of it making its way into v2 is undetermined yet slim.

**Comment by rpaggi** [comment]
What is the way to apply masks in the current version?

**Comment by andreixk** [comment]
> What is the way to apply masks in the current version?

make your own plugin or use another framework. there's no good solution for vuetify

**Comment by RonaldJerez** [comment]
I am the author of [vue-input-facade](https://github.com/RonaldJerez/vue-input-facade).  It started as a fork of vue-the-mask to fix some of the outstanding issues, but it has evolved while still maintaining a similar API.   I'll be willing to help out if you guys want to include it in vuetify.  It is currently been used in a high visible project at my current employment (Fortune 500 company), so I'll be maintaining it for quite some time.

**Comment by gogangipark** [comment]
> I am the author of [vue-input-facade](https://github.com/RonaldJerez/vue-input-facade). It started as a fork of vue-the-mask to fix some of the outstanding issues, but it has evolved while still maintaining a similar API. I'll be willing to help out if you guys want to include it in vuetify. It is currently been used in a high visible project at my current employment (Fortune 500 company), so I'll be maintaining it for quite some time.


It will be very helpful to use vuetify.

**Comment by SkyaTura** [comment]
These is any perspective on this topic? Or a recommended workaround

**Comment by builder555** [comment]
i'm using [maska](https://www.npmjs.com/package/maska). It does the job well.

**Comment by SkyaTura** [comment]
Will this ever come back to the roadmap?

**Comment by johnleider** [comment]
> Will this ever come back to the roadmap?

Yes

**Comment by MatthewAry** [comment]
Would it be possible to do an RFC on how masking will be handled?

**Comment by johnleider** [comment]
Maybe for 3.6+

**Comment by SkyaTura** [comment]
What is the official workaround recommendation? Most of the libraries made for this purpose are designed for `input` native element and does not work well with wrapped inputs like Vuetify

I understand it's a complicated feature to implement, but it is also extremely common use case of inputs overall. It would be nice to have at least a direction to look for in terms of compatibility with the framework.

**Comment by johnleider** [comment]
@KaelWD might have a recommendation for this, but I do not, sorry.

**Comment by mlebrasseur** [comment]
I use the directive from vue-input-facade and it works great. You need to use the beta version (vue 3 support).

**Comment by yuwu9145** [comment]
### Third-party integration
As per recent testing of vue-input-facade and maska. I found specific ways to make them compatible with Vuetify's v-text-field:

- For Maska, using `v-maska` directive bindings works, but applying tokens using `data-maska` doesn't work.
- For vue-input-facade, the beta version 3.0.0-beta is compatible with Vue3/Vuetify.


I have created a [demo](https://play.vuetifyjs.com/#eNqNU+9P2zAQ/Vcs86FMa2KgVJqqUrFNTGLSfmhD+0L44CbXYkgcy3YCqOr/vss50KYtVaVISe7evXv3fL5dcGdT8dmYuK6Aj/jYQ2Fy6WGSaMbG96eTH9I9yrHAL4qYSeWUnrM6KprE6JZev4xXpXZ37Km0j24sTFt+NkFcmUE+YosFa6A38OzZcomEZwFTRx5D0UxBnr3H2pJcJPyVIuFMhPqp3f5CsThOpLSpkFmmMoN0NcKGKkReN8BvhPsn8wr2CQx0KKV3FB31UMdK3A6mVudYrPmKvy61ynjmwFfBKlWY0nqGashvtmQzWxasRz70OggLMvWqhj5+zd6A2LsLCzqZdKwOetahHWuwrqlM0Wq/OqOLhv54ODz/0E2+ngoBgpLjRQNhlB4x8qUfIiDnYEeoJXdAoSWyrfh2WR/6ng+HiByLYFRrmn/JYYIxeiea97l3SDRT8/jBlRrXl4QkPC0Lo3KwrdSE40EHQQmXeV4+faeYt1VQRTX3kD7uiD+45yaW8N8WHNgaj/Qt56WdA+5ik776+5P28i2JW1HliN6T/AOuzKtGY4B9qXSGstdwpPaazhRv3Y27evag3etQjVBylfC0gF/3jL6SO4gHVJdofNDHsDV4+cyGkyHRpcEuQe6998aNhEgzjWV4B1RtYw1eaFOIS4QJW2mvCoiysrgcxOfx6SeRKefX4zG4Ipra8gnNRZa12ZHAq9nLAb1aJLYYxoPQoQ1FuZy6psUWtaDDtJEFnYFtTD9spI2yzlgbufdHo4t0QEfCXZ4FdvrZmqNzlQ/zqlOCA5zEJ9EUvIyH4iNKTvjabiz7PEyIW0L28rv/NKoTrw==) that illustrates how to integrate these two libraries with Vuetify.

It could be used as an immediate workaround if you want to try.

### Long-term plan for implementing built-in mask logic
#19207 is working in progress, with the initial goal of porting [the mask logic from v1](https://v15.vuetifyjs.com/en/components/text-fields/#masks) to v3
